### PR TITLE
Add docstring to string_like method

### PR DESCRIPTION
**Pull request by tagomatech**
Add docstring to string_like method

### CODE REVIEW

**Comment by bashtage** [comment]
There should be blank lines between sections.  You seem to have accidentally removed these.  Could you restore them in the exiting docstrings?

File: statsmodels/tools/validation/validation.py [hunk 1]
```diff
@@ -395,6 +371,27 @@ def float_like(value, name, optional=False, strict=False):
 
 
 def string_like(value, name, optional=False, options=None, lower=True):
+    """
+    Check if of type string and raise not string_like if not
```
Comment (bashtage) [inline statsmodels/tools/validation/validation.py:375]: This doesn't make much sense.  Maybe just "Check if object is string-like and raise if not"

File: statsmodels/tools/validation/validation.py [hunk 2]
```diff
@@ -395,6 +371,27 @@ def float_like(value, name, optional=False, strict=False):
...
+
+    Parameters
+    ----------
+    value : object
+        Value to verify
+    name : str
+        Variable name for exceptions
+    optional : bool
+        Flag indicating whether None is allowed
+    options : string
```
Comment (bashtage) [inline statsmodels/tools/validation/validation.py:385]: Type should be `Sequence[str]` or `tuple[str]`

Comment (tagomatech) [reply statsmodels/tools/validation/validation.py]: @bashtage I did what you suggested above
- restored blank lines between code sections
- re-phrased the docstring
- replace str with Sequence[str] or tuple[str]
Thank you very much

**Comment by bashtage** [comment]
You have to remove the white space from empty lines.

**Comment by bashtage** [comment]
LGMT. Can you ping when green? Thanks.

**Comment by tagomatech** [comment]
> LGMT. Can you ping when green? Thanks.

@bashtage 
Checks just finished. Unforfunately, **statsmodels.statsmodels — #20200817.5 failed,**
Error message: **##[error]The job running on agent Azure Pipelines 3 ran longer than the maximum time of 60 minutes.**

**Comment by bashtage** [comment]
Thanks. These happen and we ignore them.

**Comment by bashtage** [comment]
Thanks.  I did a small squash and merged in e67e6252554c1e6ed72c905bef34b2bd8f5f7ede

**Comment by bashtage** [comment]
Don't worry about the label, this is just housekeeping.

**Pull request closed without merging**